File: src/app/model/issues/IssueModelField.ts
export class IssueModelField {
  private _title: string
  private _description: string
  private _required: boolean

  constructor(title: string, description: string, required: boolean) {
    this._title = title;
    this._description = description;
    this._required = required;
  }

  get title(): string {
    return this._title;
  }

  set title(value: string) {
    this._title = value;
  }

  get description(): string {
    return this._description;
  }

  set description(value: string) {
    this._description = value;
  }

  get required(): boolean {
    return this._required;
  }

  set required(value: boolean) {
    this._required = value;
  }
}

File: src/api/src/main/java/fr/iut/uca/v1/model/issues/IssueModelField.java
package fr.iut.uca.v1.model.issues;

public class IssueModelField {

    protected String title;

    protected String description;

    protected boolean required;

    public IssueModelField(String title, String description, boolean required) throws IllegalArgumentException {
        setTitle(title);
        this.description = description;
        this.required = required;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        if (title == null || title.isBlank()) {
            throw new IllegalArgumentException("The title cannot be null or blank");
        }
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public boolean isRequired() {
        return required;
    }

    public void setRequired(boolean required) {
        this.required = required;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        IssueModelField that = (IssueModelField) o;

        return title.equals(that.title);
    }

    @Override
    public int hashCode() {
        return title.hashCode();
    }
}
